### PR TITLE
Tweaks to PIO clkdiv-setting code

### PR DESCRIPTION
**Pull request by lurch**
- refactor to reduce duplication
 - add extra param-validation

You get "unexpected results" if the divider is set too big :wink: https://github.com/micropython/micropython/issues/7025

### CODE REVIEW

**Comment by lurch** [comment]
The diff looks worse than it actually is because I swapped around the order of a couple of functions.

**Comment by kilograham** [comment]
why?

File: src/rp2_common/hardware_pio/include/hardware/pio.h [hunk 1]
```diff
@@ -238,6 +215,38 @@ static inline void sm_config_set_clkdiv_int_frac(pio_sm_config *c, uint16_t div_
             (((uint)div_int) << PIO_SM0_CLKDIV_INT_LSB);
 }
 
+static inline void calculate_clkdiv_from_float(float div, uint16_t *div_int, uint8_t *div_frac) {
```
Comment (kilograham) [inline src/rp2_common/hardware_pio/include/hardware/pio.h:218]: this is a global namespace, you should call this something like pio_calculate_clkdiv_from_float

**Review by kilograham — CHANGES_REQUESTED** [review]

File: src/rp2_common/hardware_pio/include/hardware/pio.h (hunk 1 above)
Comment (lurch) [reply src/rp2_common/hardware_pio/include/hardware/pio.h]: Ahhh, this _did_ trigger my spidey-sense, and I wondered if it should have been named `_calculate_clkdiv_from_float` to mark it as "internal use only" as you'd typically do in Python. I've now renamed it to `pio_calculate_clkdiv_from_float` as you suggested. :+1:

**Comment by lurch** [comment]
Manually "rebased" to `develop-1.1.1` and force-pushed.

**Review by kilograham — APPROVED** [review]

**Pull request merged**